src/auth/controllers/users.controller.ts

```diff
@@ -1,24 +1,85 @@
-import { Controller, Get, Param, UseGuards } from '@nestjs/common';
+import {
+  Body,
+  Controller,
+  Get,
+  Param,
+  Post,
+  Put,
+  Request,
+  UseGuards,
+} from '@nestjs/common';
 import { AuthGuard } from '@nestjs/passport';
+import { Observable } from 'rxjs';
 import { hasRoles } from '../decorators/roles.decorator';
 import { RolesGuard } from '../guards/roles.guard';
+import {
+  FriendRequest,
+  FriendRequestStatus,
+} from '../models/friend-request.interface';
 import { Role } from '../models/role.enum';
 import { User } from '../models/user.entity';
 import { UserService } from '../services/users.service';
 
-@hasRoles(Role.ADMIN)
-@UseGuards(AuthGuard(), RolesGuard)
 @Controller('users')
 export class UserController {
   constructor(private userService: UserService) {}
 
+  @hasRoles(Role.ADMIN)
+  @UseGuards(AuthGuard(), RolesGuard)
   @Get()
   getAllUsers(): Promise<User[]> {
     return this.userService.getUsers();
   }
 
+  @hasRoles(Role.ADMIN, Role.USER)
+  @UseGuards(AuthGuard(), RolesGuard)
   @Get('/:id')
-  getUserById(@Param('id') id: string): Promise<User> {
+  getUserById(@Param('id') id: number): Promise<Observable<User>> {
     return this.userService.getUserById(id);
   }
+
+  @hasRoles(Role.ADMIN, Role.USER)
+  @UseGuards(AuthGuard(), RolesGuard)
+  @Post('friend-request/send/:receiverId')
+  async sendFriendRequest(
+    @Param('receiverId') receiverStringId: string,
```

Comment:
em dùng thằng `ParseIntPipe` luôn, ko cần phải convert ở code trong hàm nữa nhé
sửa tương tự những chỗ khác
```suggestion
    @Param('receiverId', ParseIntPipe) receiverStringId: number,
```

Reply:
Dạ vâng ạ